compiler/aarch64/codegen/GenerateInstructions.cpp

```diff
@@ -350,6 +350,7 @@ TR::Instruction *generateCompareImmInstruction(TR::CodeGenerator *cg, TR::Node *
    /* Alias of SUBS instruction */
 
    TR::InstOpCode::Mnemonic op = is64bit ? TR::InstOpCode::subsimmx : TR::InstOpCode::subsimmw;
+   TR_ASSERT(constantIsUnsignedImm12(imm), "Immediate value is out of range for subsimm");
```

Comment:
Why not a fatal assert?  We're clearly using the wrong instruction for this size of immediate.

Reply:
I also need to change all the `TR_ASSERT()` lines for shift amounts in the same file, then.

Reply:
Changed them all to ` TR_ASSERT_FATAL()`.